sdk/formrecognizer/ai-form-recognizer/test/utils/recordedClients.ts

```diff
@@ -0,0 +1,111 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+import { Context } from "mocha";
+
+import {
+  env,
+  Recorder,
+  record,
+  RecorderEnvironmentSetup,
+  isPlaybackMode,
+} from "@azure-tools/test-recorder";
+
+import { AzureKeyCredential } from "../../src";
+import { ClientSecretCredential } from "@azure/identity";
+import { KeyCredential, TokenCredential } from "@azure/core-auth";
+
+export interface RecordedClient<T> {
+  client: T;
+  recorder: Recorder;
+}
+
+const replaceableVariables: { [k: string]: string } = {
+  AZURE_CLIENT_ID: "azure_client_id",
+  AZURE_CLIENT_SECRET: "azure_client_secret",
+  AZURE_TENANT_ID: "12345678-1234-1234-1234-123456789012",
+  FORM_RECOGNIZER_API_KEY: "api_key",
+  FORM_RECOGNIZER_ENDPOINT: "https://endpoint/",
+  FORM_RECOGNIZER_TRAINING_CONTAINER_SAS_URL: "https://storageaccount/trainingdata?sastoken",
+  FORM_RECOGNIZER_TESTING_CONTAINER_SAS_URL: "https://storageaccount/testingdata?sastoken",
+  FORM_RECOGNIZER_SELECTION_MARK_STORAGE_CONTAINER_SAS_URL:
+    "https://storageaccount/selectionmark-v3?sastoken",
+  FORM_RECOGNIZER_TARGET_RESOURCE_REGION: "westus2",
+  // fake resource id
+  FORM_RECOGNIZER_TARGET_RESOURCE_ID:
+    "/subscriptions/e1367d46-77d4-4f57-8cfe-348edbdc84a3/resourceGroups/jstests/providers/Microsoft.CognitiveServices/accounts/jstests-fr",
+};
+
+export const testEnv = new Proxy(replaceableVariables, {
+  get: (target, key: string) => {
+    return env[key] || target[key];
+  },
+});
```

Comment:
Could env vars be accessed from `env` directly?

Reply:
Yes, I think this is a relic, but I'm hesitant to change it now at the 11th hour. I'll do a more thorough review of the tests before beta.2.